# Development and Validation of a Risk Prediction Model of

linezolid-induced thrombocytopenia in Vietnamese patients

## Objectives

1.  Investigating risk factors of linezolid-induced thrombocytopenia (LI-TP)
2.  Developing and validating a logistics regression model to predict LI-TP in Vietnamese patients

## Data cleaning

In [ ]:
source("notebooks/initialize-data-analysis.r")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 5669 Columns: 113
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (4): patient_ID, site, dept, LZD_route
dbl  (20): patient_age, patient_weight, charlson, baseline_CLCR, baseline_WB...
lgl  (85): patient_sex, dept_ICU, dept_ER, dept_other, invasive_ETI, invasiv...
date  (4): baseline_date, LZD_start, LZD_end, test_date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column t

``` r
data_patient_complete |> select(all_of(predictor_list), flag_ADR_TP_ID, site) |> glimpse()
```

    Rows: 780
    Columns: 58
    $ patient_age          <dbl> 90, 80, 79, 71, 72, 61, 60, 64, 92, 75, 86, 93, 6…
    $ patient_sex          <lgl> TRUE, TRUE, FALSE, FALSE, TRUE, FALSE, FALSE, TRU…
    $ LZD_dose_per_weight  <dbl> 25.00000, 30.00000, 30.00000, 13.33333, 17.14286,…
    $ baseline_CLCR        <dbl> 27.22860, 63.15805, 29.93031, 50.89929, 10.87932,…
    $ dept_ER              <lgl> TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, T…
    $ dept_ICU             <lgl> FALSE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, FALSE,…
    $ baseline_HGB         <dbl> 96, 101, 86, 94, 86, 99, 98, 119, 60, 118, 99, 10…
    $ baseline_WBC         <dbl> 6.75, 11.91, 14.05, 14.61, 7.92, 21.79, 13.27, 6.…
    $ baseline_PLT         <dbl> 244, 180, 259, 179, 236, 113, 196, 154, 147, 101,…
    $ LZD_duration         <dbl> 6, 8, 15, 3, 7, 8, 22, 4, 3, 16, 14, 7, 13, 20, 6…
    $ invasive_ETI         <lgl> FALSE, FALSE, FALSE, TRUE, TRUE, FALSE, TRUE, FAL…
    $ invasive_CVC         <lgl> FALSE, FALSE, TRUE, FALSE, TRUE, FALSE, TRUE, FAL…
    $ invasive_IHD         <lgl> FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, F…
    $ invasive_CRRT        <lgl> FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, F…
    $ comorb_HTN           <lgl> TRUE, TRUE, TRUE, TRUE, TRUE, FALSE, FALSE, TRUE,…
    $ comorb_DM            <lgl> TRUE, FALSE, FALSE, FALSE, TRUE, TRUE, FALSE, FAL…
    $ comorb_HF            <lgl> FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, FALSE, TRUE…
    $ comorb_angina        <lgl> FALSE, TRUE, TRUE, FALSE, FALSE, FALSE, FALSE, FA…
    $ comorb_cirr          <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, F…
    $ comorb_COPD          <lgl> FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, F…
    $ comorb_CVA           <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comorb_MI            <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comorb_K             <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, F…
    $ comorb_hematological <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comorb_hema          <lgl> FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, F…
    $ infect_sepsis        <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, TRUE, TR…
    $ infect_CAP           <lgl> FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, FALSE, F…
    $ infect_HAP           <lgl> TRUE, FALSE, FALSE, TRUE, TRUE, FALSE, FALSE, TRU…
    $ infect_SSTI          <lgl> FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, T…
    $ infect_CNS           <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, F…
    $ infect_IAI           <lgl> FALSE, TRUE, FALSE, FALSE, FALSE, TRUE, TRUE, FAL…
    $ infect_UTI           <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ infect_BJI           <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ infect_septicemia    <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_aspirin        <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_diclofenac     <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_ibuprofen      <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_paracetamol    <lgl> TRUE, TRUE, FALSE, FALSE, TRUE, TRUE, TRUE, TRUE,…
    $ comed_penicillin     <lgl> FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, FALSE…
    $ comed_cepha          <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, F…
    $ comed_carbapenem     <lgl> TRUE, TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, …
    $ comed_cotrimoxazol   <lgl> FALSE, TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, F…
    $ comed_vancomycin     <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_levofloxacin   <lgl> FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, F…
    $ comed_teicoplanin    <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_ethambutol     <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_pyrazinamid    <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_rifampin       <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_heparin        <lgl> FALSE, FALSE, FALSE, TRUE, TRUE, FALSE, FALSE, FA…
    $ comed_clopidogrel    <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_enoxaparin     <lgl> FALSE, FALSE, TRUE, TRUE, FALSE, TRUE, TRUE, FALS…
    $ comed_dexamethason   <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, FALSE, F…
    $ comed_amiodaron      <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_furosemid      <lgl> FALSE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, FALSE,…
    $ comed_haloperidol    <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ comed_valproic       <lgl> FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, …
    $ flag_ADR_TP_ID       <lgl> FALSE, FALSE, FALSE, FALSE, TRUE, TRUE, TRUE, FAL…
    $ site                 <chr> "TN1", "TN1", "TN1", "TN1", "TN1", "TN1", "TN1", …

## Descriptive statistics

In [ ]:
load("data/results/table-descriptive.rda")

``` r
table_descriptive
```

<div id="ujuebtwdih" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ujuebtwdih table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ujuebtwdih thead, #ujuebtwdih tbody, #ujuebtwdih tfoot, #ujuebtwdih tr, #ujuebtwdih td, #ujuebtwdih th {
  border-style: none;
}

#ujuebtwdih p {
  margin: 0;
  padding: 0;
}

#ujuebtwdih .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ujuebtwdih .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ujuebtwdih .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ujuebtwdih .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ujuebtwdih .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ujuebtwdih .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ujuebtwdih .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ujuebtwdih .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ujuebtwdih .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ujuebtwdih .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ujuebtwdih .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ujuebtwdih .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ujuebtwdih .gt_spanner_row {
  border-bottom-style: hidden;
}

#ujuebtwdih .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ujuebtwdih .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ujuebtwdih .gt_from_md > :first-child {
  margin-top: 0;
}

#ujuebtwdih .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ujuebtwdih .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ujuebtwdih .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ujuebtwdih .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ujuebtwdih .gt_row_group_first td {
  border-top-width: 2px;
}

#ujuebtwdih .gt_row_group_first th {
  border-top-width: 2px;
}

#ujuebtwdih .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ujuebtwdih .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ujuebtwdih .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ujuebtwdih .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ujuebtwdih .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ujuebtwdih .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ujuebtwdih .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ujuebtwdih .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ujuebtwdih .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ujuebtwdih .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ujuebtwdih .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ujuebtwdih .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ujuebtwdih .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ujuebtwdih .gt_left {
  text-align: left;
}

#ujuebtwdih .gt_center {
  text-align: center;
}

#ujuebtwdih .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ujuebtwdih .gt_font_normal {
  font-weight: normal;
}

#ujuebtwdih .gt_font_bold {
  font-weight: bold;
}

#ujuebtwdih .gt_font_italic {
  font-style: italic;
}

#ujuebtwdih .gt_super {
  font-size: 65%;
}

#ujuebtwdih .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ujuebtwdih .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ujuebtwdih .gt_indent_1 {
  text-indent: 5px;
}

#ujuebtwdih .gt_indent_2 {
  text-indent: 10px;
}

#ujuebtwdih .gt_indent_3 {
  text-indent: 15px;
}

#ujuebtwdih .gt_indent_4 {
  text-indent: 20px;
}

#ujuebtwdih .gt_indent_5 {
  text-indent: 25px;
}
</style>

  -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  **Characteristic**                                                  [**Thrombocytopenia Status**]{.gt_column_spanner}                                                                                                                                                           [**Univariate Regression**]{.gt_column_spanner}                                                                                         
  ------------------------------------------------------------------- ------------------------------------------------------------------- ------------------------------------------------------------------- ------------------------------------------------------------------- ------------------------------------------------------------------- ------------------------------------------------------------------- -------------
                                                                      **Overall**, N = 780[^1^]{.gt_footnote_marks                        **FALSE**, N = 520[^1^]{.gt_footnote_marks                          **TRUE**, N = 260[^1^]{.gt_footnote_marks                           **OR**[^2^]{.gt_footnote_marks                                      **95% CI**[^2^]{.gt_footnote_marks                                  **p-value**
                                                                      style="white-space:nowrap;font-style:italic;font-weight:normal;"}   style="white-space:nowrap;font-style:italic;font-weight:normal;"}   style="white-space:nowrap;font-style:italic;font-weight:normal;"}   style="white-space:nowrap;font-style:italic;font-weight:normal;"}   style="white-space:nowrap;font-style:italic;font-weight:normal;"}   

  patient_age                                                         62 (50 - 73)                                                        61 (48 - 72)                                                        64 (53 - 74)                                                        1.02                                                                1.01, 1.02                                                          \<0.001

  patient_sex                                                         292 (37%)                                                           194 (37%)                                                           98 (38%)                                                            1.02                                                                0.75, 1.38                                                          \>0.9

  LZD_dose_per_weight                                                 21.8 (20.0 - 24.0)                                                  21.8 (20.0 - 24.0)                                                  21.8 (19.4 - 24.5)                                                  0.99                                                                0.95, 1.03                                                          0.5

  baseline_CLCR                                                       46 (21 - 83)                                                        55 (26 - 88)                                                        32 (15 - 64)                                                        0.99                                                                0.99, 0.99                                                          \<0.001

  dept_ER                                                             133 (17%)                                                           89 (17%)                                                            44 (17%)                                                            0.99                                                                0.66, 1.46                                                          \>0.9

  dept_ICU                                                            368 (47%)                                                           221 (43%)                                                           147 (57%)                                                           1.76                                                                1.30, 2.38                                                          \<0.001

  baseline_HGB                                                        102 (89 - 119)                                                      105 (91 - 121)                                                      97 (85 - 117)                                                       0.99                                                                0.98, 1.0                                                           \<0.001

  baseline_WBC                                                        12 (8 - 17)                                                         12 (8 - 17)                                                         12 (8 - 18)                                                         1.01                                                                0.99, 1.03                                                          0.3

  baseline_PLT                                                        203 (141 - 286)                                                     233 (165 - 310)                                                     151 (102 - 208)                                                     0.99                                                                0.99, 0.99                                                          \<0.001

  LZD_duration                                                        9.0 (6.0 - 14.0)                                                    9.0 (6.0 - 13.0)                                                    10.0 (6.0 - 14.0)                                                   1.03                                                                1.01, 1.06                                                          0.016

  invasive_ETI                                                        363 (47%)                                                           210 (40%)                                                           153 (59%)                                                           2.11                                                                1.56, 2.86                                                          \<0.001

  invasive_CVC                                                        399 (51%)                                                           226 (43%)                                                           173 (67%)                                                           2.59                                                                1.90, 3.54                                                          \<0.001

  invasive_IHD                                                        105 (13%)                                                           60 (12%)                                                            45 (17%)                                                            1.60                                                                1.05, 2.44                                                          0.027

  invasive_CRRT                                                       133 (17%)                                                           53 (10%)                                                            80 (31%)                                                            3.92                                                                2.67, 5.79                                                          \<0.001

  comorb_HTN                                                          321 (41%)                                                           208 (40%)                                                           113 (43%)                                                           1.15                                                                0.85, 1.56                                                          0.4

  comorb_DM                                                           214 (27%)                                                           142 (27%)                                                           72 (28%)                                                            1.02                                                                0.73, 1.42                                                          \>0.9

  comorb_HF                                                           218 (28%)                                                           126 (24%)                                                           92 (35%)                                                            1.71                                                                1.24, 2.37                                                          0.001

  comorb_angina                                                       31 (4.0%)                                                           19 (3.7%)                                                           12 (4.6%)                                                           1.28                                                                0.59, 2.64                                                          0.5

  comorb_cirr                                                         48 (6.2%)                                                           20 (3.8%)                                                           28 (11%)                                                            3.02                                                                1.67, 5.54                                                          \<0.001

  comorb_COPD                                                         38 (4.9%)                                                           24 (4.6%)                                                           14 (5.4%)                                                           1.18                                                                0.58, 2.28                                                          0.6

  comorb_CVA                                                          90 (12%)                                                            64 (12%)                                                            26 (10%)                                                            0.79                                                                0.48, 1.27                                                          0.3

  comorb_MI                                                           20 (2.6%)                                                           15 (2.9%)                                                           5 (1.9%)                                                            0.66                                                                0.21, 1.73                                                          0.4

  comorb_K                                                            64 (8.2%)                                                           41 (7.9%)                                                           23 (8.8%)                                                           1.13                                                                0.66, 1.92                                                          0.6

  comorb_hematological                                                46 (5.9%)                                                           27 (5.2%)                                                           19 (7.3%)                                                           1.44                                                                0.77, 2.63                                                          0.2

  comorb_hema                                                         60 (7.7%)                                                           36 (6.9%)                                                           24 (9.2%)                                                           1.37                                                                0.79, 2.33                                                          0.3

  infect_sepsis                                                       130 (17%)                                                           62 (12%)                                                            68 (26%)                                                            2.62                                                                1.78, 3.84                                                          \<0.001

  infect_CAP                                                          112 (14%)                                                           66 (13%)                                                            46 (18%)                                                            1.48                                                                0.98, 2.22                                                          0.062

  infect_HAP                                                          352 (45%)                                                           236 (45%)                                                           116 (45%)                                                           0.97                                                                0.72, 1.31                                                          0.8

  infect_SSTI                                                         128 (16%)                                                           95 (18%)                                                            33 (13%)                                                            0.65                                                                0.42, 0.99                                                          0.049

  infect_CNS                                                          68 (8.7%)                                                           45 (8.7%)                                                           23 (8.8%)                                                           1.02                                                                0.60, 1.72                                                          \>0.9

  infect_IAI                                                          49 (6.3%)                                                           33 (6.3%)                                                           16 (6.2%)                                                           0.97                                                                0.51, 1.77                                                          \>0.9

  infect_UTI                                                          53 (6.8%)                                                           37 (7.1%)                                                           16 (6.2%)                                                           0.86                                                                0.46, 1.54                                                          0.6

  infect_BJI                                                          10 (1.3%)                                                           9 (1.7%)                                                            1 (0.4%)                                                            0.22                                                                0.01, 1.18                                                          0.2

  infect_septicemia                                                   231 (30%)                                                           143 (28%)                                                           88 (34%)                                                            1.35                                                                0.98, 1.86                                                          0.068

  comed_aspirin                                                       46 (5.9%)                                                           29 (5.6%)                                                           17 (6.5%)                                                           1.18                                                                0.63, 2.17                                                          0.6

  comed_diclofenac                                                    27 (3.5%)                                                           20 (3.8%)                                                           7 (2.7%)                                                            0.69                                                                0.27, 1.58                                                          0.4

  comed_ibuprofen                                                     25 (3.2%)                                                           14 (2.7%)                                                           11 (4.2%)                                                           1.60                                                                0.70, 3.56                                                          0.3

  comed_paracetamol                                                   338 (43%)                                                           230 (44%)                                                           108 (42%)                                                           0.90                                                                0.66, 1.21                                                          0.5

  comed_penicillin                                                    114 (15%)                                                           71 (14%)                                                            43 (17%)                                                            1.25                                                                0.83, 1.88                                                          0.3

  comed_cepha                                                         197 (25%)                                                           141 (27%)                                                           56 (22%)                                                            0.74                                                                0.52, 1.05                                                          0.092

  comed_carbapenem                                                    554 (71%)                                                           355 (68%)                                                           199 (77%)                                                           1.52                                                                1.08, 2.14                                                          0.017

  comed_cotrimoxazol                                                  63 (8.1%)                                                           36 (6.9%)                                                           27 (10%)                                                            1.56                                                                0.92, 2.62                                                          0.10

  comed_vancomycin                                                    67 (8.6%)                                                           40 (7.7%)                                                           27 (10%)                                                            1.39                                                                0.83, 2.31                                                          0.2

  comed_levofloxacin                                                  230 (29%)                                                           144 (28%)                                                           86 (33%)                                                            1.29                                                                0.93, 1.78                                                          0.12

  comed_teicoplanin                                                   26 (3.3%)                                                           13 (2.5%)                                                           13 (5.0%)                                                           2.05                                                                0.93, 4.54                                                          0.072

  comed_ethambutol                                                    8 (1.0%)                                                            5 (1.0%)                                                            3 (1.2%)                                                            1.20                                                                0.25, 4.94                                                          0.8

  comed_pyrazinamid                                                   12 (1.5%)                                                           6 (1.2%)                                                            6 (2.3%)                                                            2.02                                                                0.63, 6.53                                                          0.2

  comed_rifampin                                                      17 (2.2%)                                                           10 (1.9%)                                                           7 (2.7%)                                                            1.41                                                                0.51, 3.72                                                          0.5

  comed_heparin                                                       189 (24%)                                                           94 (18%)                                                            95 (37%)                                                            2.61                                                                1.86, 3.66                                                          \<0.001

  comed_clopidogrel                                                   38 (4.9%)                                                           28 (5.4%)                                                           10 (3.8%)                                                           0.70                                                                0.32, 1.42                                                          0.3

  comed_enoxaparin                                                    327 (42%)                                                           213 (41%)                                                           114 (44%)                                                           1.13                                                                0.83, 1.52                                                          0.4

  comed_dexamethason                                                  92 (12%)                                                            60 (12%)                                                            32 (12%)                                                            1.08                                                                0.67, 1.69                                                          0.8

  comed_amiodaron                                                     33 (4.2%)                                                           15 (2.9%)                                                           18 (6.9%)                                                           2.50                                                                1.24, 5.12                                                          0.010

  comed_furosemid                                                     417 (53%)                                                           244 (47%)                                                           173 (67%)                                                           2.25                                                                1.65, 3.08                                                          \<0.001

  comed_haloperidol                                                   45 (5.8%)                                                           28 (5.4%)                                                           17 (6.5%)                                                           1.23                                                                0.65, 2.27                                                          0.5

  comed_valproic                                                      29 (3.7%)                                                           21 (4.0%)                                                           8 (3.1%)                                                            0.75                                                                0.31, 1.66                                                          0.5

  comed_aceclofenac                                                   0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_naproxen                                                      0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_daptomycin                                                    1 (0.1%)                                                            0 (0%)                                                              1 (0.4%)                                                            \                                                                   \                                                                   \

  comed_cetirizin                                                     6 (0.8%)                                                            5 (1.0%)                                                            1 (0.4%)                                                            \                                                                   \                                                                   \

  comed_simvas                                                        0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_bisoprolol                                                    6 (0.8%)                                                            4 (0.8%)                                                            2 (0.8%)                                                            \                                                                   \                                                                   \

  comed_diltiazem                                                     0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_eptifibatid                                                   0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_quinidin                                                      0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_carbamazepin                                                  7 (0.9%)                                                            7 (1.3%)                                                            0 (0%)                                                              \                                                                   \                                                                   \

  comed_phenytoin                                                     0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_mirtazapin                                                    0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_quetiapin                                                     3 (0.4%)                                                            3 (0.6%)                                                            0 (0%)                                                              \                                                                   \                                                                   \

  comed_ondansetron                                                   6 (0.8%)                                                            4 (0.8%)                                                            2 (0.8%)                                                            \                                                                   \                                                                   \

  comed_palonosetron                                                  0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_oseltamivir                                                   3 (0.4%)                                                            1 (0.2%)                                                            2 (0.8%)                                                            \                                                                   \                                                                   \

  comed_quinin                                                        0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_pembrolizumab                                                 0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_trastuzumab                                                   0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_atezolizumab                                                  0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_durvalumab                                                    0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_IVIG                                                          0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_tacrolimus                                                    1 (0.1%)                                                            0 (0%)                                                              1 (0.4%)                                                            \                                                                   \                                                                   \

  comed_fluorouracil                                                  0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_irinotecan                                                    0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_leucovorin                                                    0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  comed_oxaliplatin                                                   0 (0%)                                                              0 (0%)                                                              0 (0%)                                                              \                                                                   \                                                                   \

  [^1^]{.gt_footnote_marks                                                                                                                                                                                                                                                                                                                                                                                                
  style="white-space:nowrap;font-style:italic;font-weight:normal;"}                                                                                                                                                                                                                                                                                                                                                       
  Median (IQR); n (%)                                                                                                                                                                                                                                                                                                                                                                                                     

  [^2^]{.gt_footnote_marks                                                                                                                                                                                                                                                                                                                                                                                                
  style="white-space:nowrap;font-style:italic;font-weight:normal;"}                                                                                                                                                                                                                                                                                                                                                       
  OR = Odds Ratio, CI = Confidence Interval                                                                                                                                                                                                                                                                                                                                                                               
  -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

</div>

## Model Performance

In [ ]:
load("data/model-performance/apparent-performance-metrics.rda")
load("data/model-performance/corrected-performance-metrics.rda")
load("data/model-performance/fold-performance-metrics.rda")
load("data/model-performance/cross-performance-metrics.rda")

data_list <- list(
  apparent_performance_metrics,
  corrected_performance_metrics,
  fold_performance_metrics
)
performance_type_list <- c("Apparent", "Bootstrap", "Fold")
initial_string_list <- c("apparent_", "corrected_", "mean_fold_")

reformat_performance_metrics <- function(data, type, string) {
  data |> 
    mutate(
    performance_type = type
  ) |>
  rename_with(~ str_replace(., string, "")) |>
  select(performance_type, everything())
}

performance_metrics <- pmap(
  list(data_list, performance_type_list, initial_string_list), 
  reformat_performance_metrics
) |> bind_rows()

performance_metrics |> knitr::kable()

  performance_type       C_index   calibration_intercept   calibration_slope
  ------------------ ----------- ----------------------- -------------------
  Apparent             0.7779549               0.0000000           1.0000000
  Bootstrap            0.7456938              -0.0001972           0.8176676
  Fold                 0.7641578              -0.0084625           0.9584548
